integration-testing/src/main/java/io/grpc/testing/integration/TestServiceImpl.java

```diff
@@ -293,7 +294,14 @@ private void scheduleNextChunk() {
         Chunk nextChunk = chunks.peek();
         if (nextChunk != null) {
           scheduled = true;
-          executor.schedule(dispatchTask, nextChunk.delayMicroseconds, TimeUnit.MICROSECONDS);
+          if (executor instanceof ScheduledExecutorService) {
```

Comment:
Hmm... my condition was a bit off. I'm concerned about the delay not happening.

How about something more along these lines:
if (nextChunk.delayMicroseconds != 0) {
  ((ScheduledExecutorService) executor).schedule(
      dispatchTask, nextChunk.delayMicroseconds, TimeUnit.MICROSECONDS);
} else {
  dispatchTask.run();
}